src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -968,10 +952,11 @@ internal static bool SortedSetEquals(SortedSet<T> set1, SortedSet<T> set2, IComp
 
         }
 
-        // This is a little frustrating because we can't support more sorted structures.
-        private static bool AreComparersEqual(SortedSet<T> set1, SortedSet<T> set2)
+        private bool HasEqualComparer(SortedSet<T> other)
```

Comment:
In nearly all of the places this method is being used, `this` is passed as a parameter. This makes it more concise. I also think it's better to have instance methods when you can because then you don't have to worry about null in the callee.